File: app/src/main/java/com/kunkunapp/allvideodowloader/views/NotificationBadge.java
package com.kunkunapp.allvideodowloader.views;

import android.view.LayoutInflater;
import android.view.View;

import com.google.android.material.bottomnavigation.BottomNavigationItemView;
import com.google.android.material.bottomnavigation.BottomNavigationMenuView;
import com.google.android.material.bottomnavigation.BottomNavigationView;
import com.kunkunapp.allvideodowloader.R;

public class NotificationBadge {
    public static Badge getBadge(BottomNavigationView bottomNavView, int indexAt) {
        BottomNavigationMenuView bottomNavigationMenuView =
                (BottomNavigationMenuView) bottomNavView.getChildAt(0);
        if (bottomNavigationMenuView.getChildCount() > indexAt) {
            View v = bottomNavigationMenuView.getChildAt(indexAt);
            BottomNavigationItemView itemView = (BottomNavigationItemView) v;
            if (itemView.findViewById(R.id.notifications_nav_badge) == null) {
                View badge = LayoutInflater.from(bottomNavigationMenuView.getContext()).inflate(R.layout.bottom_tab_layout, itemView, true);
                return badge.findViewById(R.id.notifications_nav_badge);
            } else {
                return itemView.findViewById(R.id.notifications_nav_badge);
            }

        } else {
            throw new IndexOutOfBoundsException(String.format("No menu item at %d", indexAt));
        }
    }
}

File: app/src/main/java/com/cyberIyke/allvideodowloader/views/NotificationBadge.kt
package com.cyberIyke.allvideodowloader.views

import android.view.LayoutInflater
import android.view.View
import com.cyberIyke.allvideodowloader.R
import com.google.android.material.bottomnavigation.BottomNavigationItemView
import com.google.android.material.bottomnavigation.BottomNavigationMenuView
import com.google.android.material.bottomnavigation.BottomNavigationView

object NotificationBadge {
    fun getBadge(bottomNavView: BottomNavigationView, indexAt: Int): Badge {
        val bottomNavigationMenuView = bottomNavView.getChildAt(0) as BottomNavigationMenuView
        return if (bottomNavigationMenuView.childCount > indexAt) {
            val v = bottomNavigationMenuView.getChildAt(indexAt)
            val itemView = v as BottomNavigationItemView
            if (itemView.findViewById<View?>(R.id.notifications_nav_badge) == null) {
                val badge = LayoutInflater.from(bottomNavigationMenuView.context)
                    .inflate(R.layout.bottom_tab_layout, itemView, true)
                badge.findViewById(R.id.notifications_nav_badge)
            } else {
                itemView.findViewById(R.id.notifications_nav_badge)
            }
        } else {
            throw IndexOutOfBoundsException(String.format("No menu item at %d", indexAt))
        }
    }
}